src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -176,7 +176,13 @@ public DefaultLogger(OutputStream infoStream,
                          OutputStream errorStream,
                          OutputStreamOptions errorStreamOptions,
                          AuditEventFormatter messageFormatter) {
+        if (infoStreamOptions == null) {
+            throw new IllegalArgumentException("Parameter infoStreamOptions can not be null");
```

Comment:
Just curious, why are we adding these IAEs and do we need them in other areas?

Reply:
Pitest. It says "the result may be replaced with `null` here
```
    private static AutomaticBean.OutputStreamOptions getOutputStreamOptions(String outputLocation) {
        final AutomaticBean.OutputStreamOptions result;
        if (outputLocation == null) {
            result = AutomaticBean.OutputStreamOptions.NONE;  // THIS line was red
        }
        else {
            result = AutomaticBean.OutputStreamOptions.CLOSE;
        }
        return result;
    }
```

The expression
```
closeStream = outputStreamOptions == OutputStreamOptions.CLOSE;
```

work just fine when the parameter `outputStreamOptions` is `null`. 
It turns out that we can use an enumeration with just one value! The second value will be `null`.
This is logical, but wrong. I explicitly forbid `null` for these parameters.